sql/core/src/test/scala/org/apache/spark/sql/connector/GroupBasedUpdateTableSuite.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.expressions.DynamicPruningExpression
+import org.apache.spark.sql.execution.{InSubqueryExec, ReusedSubqueryExec}
+import org.apache.spark.sql.execution.datasources.v2.BatchScanExec
+import org.apache.spark.sql.internal.SQLConf
+
+class GroupBasedUpdateTableSuite extends UpdateTableSuiteBase {
+
+  import testImplicits._
+
+  test("update runtime group filtering (DPP enabled)") {
```

Comment:
Like the following, please merge two test case to remove repetition.
https://github.com/apache/spark/blob/fff0b71d24bf179ccf4f9fcd7a095dcafe19a77b/sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala#L1162-L1163

Reply:
Sounds good, I'll update DELETE and MERGE tests in a separate PR for consistency.